Sources/MapboxMaps/Ornaments/Compass/MapboxCompassOrnamentView.swift

```diff
@@ -51,19 +51,25 @@ internal class MapboxCompassOrnamentView: UIButton {
                                               comment: "Accessibility hint")
 
         if let image = createCompassImage() {
-            containerView.image = image
-            NSLayoutConstraint.activate([
-                widthAnchor.constraint(equalToConstant: image.size.width),
-                heightAnchor.constraint(equalToConstant: image.size.height),
-                containerView.widthAnchor.constraint(equalToConstant: image.size.width),
-                containerView.heightAnchor.constraint(equalToConstant: image.size.height)
-            ])
+            updateImage(image: image)
         }
         containerView.translatesAutoresizingMaskIntoConstraints = false
         addSubview(containerView)
         addTarget(self, action: #selector(didTap), for: .touchUpInside)
     }
 
+    func updateImage(image: UIImage?) {
+        if image != nil {
+            containerView.image = image
+            NSLayoutConstraint.activate([
+                widthAnchor.constraint(equalToConstant: image!.size.width),
+                heightAnchor.constraint(equalToConstant: image!.size.height),
+                containerView.widthAnchor.constraint(equalToConstant: image!.size.width),
+                containerView.heightAnchor.constraint(equalToConstant: image!.size.height)
+            ])
+        }
```

Comment:
It is better to use `guard` statement to unwrap optional values:
```suggestion
        guard let image = image else { return }

        containerView.image = image

        NSLayoutConstraint.activate([
            widthAnchor.constraint(equalToConstant: image.size.width),
            heightAnchor.constraint(equalToConstant: image.size.height),
            containerView.widthAnchor.constraint(equalToConstant: image.size.width),
            containerView.heightAnchor.constraint(equalToConstant: image.size.height)
        ])
```